### ISSUE TITLE
Refactor Rename doesn't work for simple type rename

### ISSUE THREAD

**Issue by royalstream** [dotnet/fsharp#13199]
Consider this simple example

```F#
type MyType() = 
    member x.DoNothing(d:MyType) = ()

let x = MyType() // alternatively you can write: new MyType(), but refactoring won't work either
x.DoNothing(x)
```

Refactoring simply doesn't work. 
I reproduced this bug in the latest versions of VS2019 and VS2022.

**Comment by dsyme** [comment]
Actually I tried to repro in latest VS2022 and the rename refactor worked just fine.  Are you sure it's not working for you?

https://user-images.githubusercontent.com/7204669/170979014-423b9924-3ac0-4be9-8c13-289b257693aa.mp4

**Comment by royalstream** [comment]
Yes, that's exactly what's happening, it's considering the (unit -> MyType) function to be unrelated.

For some reason the refactoring works correctly if it's a script:

<img width="354" alt="image" src="https://user-images.githubusercontent.com/7918404/171025074-d0efb0d4-9b2a-4359-a2d6-8eb89273fee2.png">

But it fails if it's an ordinary **.fs** inside a project:

<img width="359" alt="image" src="https://user-images.githubusercontent.com/7918404/171025126-61ccb0d9-0ac5-4fae-9e1c-8e60b4a4e454.png">

Furthermore, in the screenshot above it's only missing the construction without the **new** keyword. In my initial test it was missing both. 
To reproduce this behavior it's a matter of removing the **new** keyword, restarting Visual Studio and waiting for the project to fully load. Then add the **new** keyword back and do the refactoring, it will get missed this time around:

<img width="349" alt="image" src="https://user-images.githubusercontent.com/7918404/171027034-c5a3b66f-5d69-4600-a798-b7349cfd19a7.png">

**Comment by T-Gro** [comment]
FindAllReferences behaves differently than GetUsesOfSymbolInFile, it is not even symmetric.
FindAllReferences (what powers rename), when applied on the 4 occurences of this sample, always returns something else.

Depending on the starting point used:

Type definiton -> finds method argument and "b"
"b" -> finds "a"
"a" -> finds "b"

**Comment by vzarytovskii** [comment]
@0101 works on renaming right now, there are some different paths when used on type definitions, or on usages.
Probably makes sense to sync with him

**Comment by T-Gro** [comment]
Yep we did yesterday.
Right now it all hints towards the "optimized" code path for Rename (and also find references) not returning all hits.

The naive way of making things correct would make them slow, so it will get tricky :((

**Comment by 0101** [comment]
I didn't really dig deep into this, but both _Rename_ and _Find all references_ use `SymbolHelpers.getSymbolUsesInProjects` as far as I can tell. Also give the same results for the repro code here. (Except when you use _Find all references_ that doesn't include the occurrence where you issued the command, but I suppose that's expected.)

And that one then calls `FSharpChecker.FindBackgroundReferencesInFile`, so the issue must be there. Probably in how the symbols uses are stored in the `ItemKeyStore`. 

@T-Gro not sure which "optimized" code path you mean, but [this one](https://github.com/dotnet/fsharp/pull/14293) hasn't been merged (and also behaves identically in this case).

**Comment by T-Gro** [comment]
Optimized code path for FindBackgroundReferencesInFile which then uses ItemKey.fs and binary blobs.
